lib/SimpleSAML/Utils/HTTP.php

```diff
@@ -85,6 +85,10 @@ private static function getServerHTTPS()
             return false;
         }
 
+        if ($_SERVER['HTTPS'] === null) {
```

Comment:
What's exactly the bug here?

If `$_SERVER['HTTPS']` is null, there are two possible reasons for that:

* It is undefined, in which case, the previous code will get rid of it (`if (!array_key_exists('HTTPS', $_SERVER)) { return false; }`).
* It is explicitly defined to `null`, which is not something PHP is going to do itself, and therefore it must be someone, somewhere, doing it explicitly.